framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -101,9 +103,23 @@
                 infer(typeFactory, argTypes, assignedTo, methodElem, methodType, targets);
 
         handleUninferredTypeVariables(typeFactory, methodType, targets, inferredArgs);
+        // Type arguments cannot be primitives, so box them if they are.
+        boxPrimitives(typeFactory, inferredArgs);
```

Comment:
Instead of post-processing the map here, would it be easier to directly incorporate this into `org.checkerframework.framework.util.typeinference.solver.InferenceResult.toAtmMap()`?
It looks lite toAtmMap is always called before `infer` returns. You would also need to make sure that `handleUninferredTypeVariables` doesn't introduce primitives, but I think that won't happen.